arroyo/processing/processor.py

```diff
@@ -132,7 +132,7 @@ def run(self) -> None:
 
             self._shutdown()
         except Exception as error:
-            logger.info("Caught %r, shutting down...", error)
+            logger.exception("Caught exception, shutting down...")
```

Comment:
We should not need to log this as an exception here as we are raising it again at line 145. What was the issue ?

Reply:
for some reason i don't fully understand, `sys.excepthook` is broken in the child process. i suspect that multiprocessing.Pool overwrites it to propagate the exception to the parent process.

for some _other_ reason, the exception that makes the parent process shut down is also not reported.

now i can choose where i want to instrument: do I want to fix instrumentation in the parent process, where the exception stacktrace is broken and local variables are missing (because multiprocessing is pickling the exception to make it cross process boundaries), or in the child process where i risk losing errors if the consumer crash-loops in the parent process?

our answer for celery integration is that we instrument in the child process because there we get better contextual data.

I think we might want to instrument both processes just to be safe, but it would inherently mean duplicated error reports, unless we remove the `raise` in the child process